### ISSUE TITLE
pip install pinned versions not recognized with `--build`, `--prefix`, `--root`, and `--target`

### ISSUE THREAD

**Issue by evolutics** [hadolint/hadolint#376]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

Related issues: #67, #115, #157, #287.

### Expected behavior

The following instructions should be fine with hadolint in terms of version pinning in pip:

- `RUN pip3 install --build /opt/yamllint yamllint==1.20.0`
- `RUN pip3 install --prefix /opt/yamllint yamllint==1.20.0`
- `RUN pip3 install --root /opt/yamllint yamllint==1.20.0`
- `RUN pip3 install --target /opt/yamllint yamllint==1.20.0`

### Actual behavior

However, there is an error
```
/dev/stdin:2 DL3013 Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>`
```

### Steps to reproduce the behavior

Use Bash to run

```bash
cat << EOF | docker run -i --rm hadolint/hadolint:v1.17.3
FROM alpine:3.11.0
RUN apk add --no-cache python3~=3.8.0 \
  && pip3 install --target /opt/yamllint yamllint==1.20.0
EOF
```

For the other cases, replace `--target` by `--build`, `--prefix`, or `--root`, respectively.

**Output of `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Unable to find image 'hadolint/hadolint:latest' locally
latest: Pulling from hadolint/hadolint
601624ad4577: Pull complete 
Digest: sha256:58cef081081977c24a0f31da95b84699a3fd981b01e54dc9e9131999f9429a9f
Status: Downloaded newer image for hadolint/hadolint:latest
Haskell Dockerfile Linter v1.17.3-3-gae8601e
```

**Dockerfiles**

```Dockerfile
FROM alpine:3.11.0
RUN apk add --no-cache python3~=3.8.0 \
  && pip3 install --build /opt/yamllint yamllint==1.20.0
```

```Dockerfile
FROM alpine:3.11.0
RUN apk add --no-cache python3~=3.8.0 \
  && pip3 install --prefix /opt/yamllint yamllint==1.20.0
```

```Dockerfile
FROM alpine:3.11.0
RUN apk add --no-cache python3~=3.8.0 \
  && pip3 install --root /opt/yamllint yamllint==1.20.0
```

```Dockerfile
FROM alpine:3.11.0
RUN apk add --no-cache python3~=3.8.0 \
  && pip3 install --target /opt/yamllint yamllint==1.20.0
```

**Additional environment details (OS, stack version, etc.)**

Xubuntu 18.04.3 LTS with Docker version 19.03.5.
